File: android/app/src/main/java/com/djs66256/short_drama/data/dto/DramaDto.kt
package com.djs66256.short_drama.data.dto

import com.djs66256.short_drama.domain.model.Drama
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DramaDto(
    val id: String,
    val title: String,
    val description: String,
    @SerialName("cover_url")
    val coverUrl: String? = null,
    val category: String,
    @SerialName("episode_count")
    val episodeCount: Int,
    val tags: List<String> = emptyList(),
    val rating: Double? = null,
    @SerialName("created_at")
    val createdAt: String,
    @SerialName("updated_at")
    val updatedAt: String
) {
    fun toDomain(): Drama = Drama(
        id = id,
        title = title,
        description = description,
        coverUrl = coverUrl.orEmpty(),
        category = category,
        episodeCount = episodeCount,
        tags = tags,
        rating = rating ?: 0.0,
        createdAt = createdAt,
        updatedAt = updatedAt
    )
}

File: ios/ShortDrama/Sources/Data/DTOs/DramaDTO.swift
import Foundation

/// API response DTO for drama data.
struct DramaDTO: Codable, Equatable {
    let id: String
    let title: String
    let description: String
    let coverUrl: String
    let category: String
    let episodeCount: Int
    let tags: [String]?
    let rating: Double?
    let createdAt: String
    let updatedAt: String
}

extension DramaDTO {

    /// Converts this DTO to a domain entity.
    func toEntity() -> Drama {
        Drama(
            id: id,
            title: title,
            description: description,
            coverUrl: coverUrl,
            category: category,
            episodeCount: episodeCount,
            tags: tags,
            rating: rating,
            createdAt: createdAt,
            updatedAt: updatedAt
        )
    }
}
